crates/bevy_window/src/event.rs

```diff
@@ -35,8 +34,21 @@ pub struct WindowCloseRequested {
     pub id: WindowId,
 }
 
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
+pub enum Cursor {
+    Vertical(WindowId),
+    Horizontal(WindowId),
+}
 #[derive(Debug, Clone)]
 pub struct CursorMoved {
```

Comment:
Splitting out the event like this makes it much harder to consume. In almost every case, users (and engine devs) will want to consume x _and_ y at the same time. Splitting them means that getting the original x,y requires reading an event, checking that its "x", then reading the next event and assuming it will be "y".

I think this should be reverted.

Reply:
So Axis should exist only as 2D?

Reply:
At the very least, I don't think we should be breaking up 2d events to meet the needs of the Axis api. I personally think 2d devices should have 2d axis apis because that will make them simpler and more natural to consume.